org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/IBuildSupport.java

```diff
@@ -70,13 +73,32 @@ default void refresh(IResource resource, CHANGE_TYPE changeType, IProgressMonito
 			return;
 		}
 		if (changeType == CHANGE_TYPE.DELETED) {
+			if (IJavaProject.CLASSPATH_FILE_NAME.equals(resource.getName())) {
+				IProject project = resource.getProject();
+				if (ProjectUtils.isJavaProject(project)) {
+					ProjectUtils.removeJavaNatureAndBuilder(project, monitor);
+					if (ProjectUtils.isMavenProject(project)) {
```

Comment:
please leave build specific processing to build specific IBuildSupport implementations